### DISCUSSION TITLE
Advice for using Quarto with older versions of Rstudio and R

### DISCUSSION THREAD

**Discussion by johnghill** [quarto-dev/quarto-cli#4110] [Q&A]
I was hoping to use quarto in a classroom setting but the PC lab is locked to the following out-dated software versions for the remainder of the academic year:

- Rstudio 2022.02.3 Build 492
- R 4.2.1 (2022-06-23 ucrt)

Installing quarto from RStudio's package manager results in a complaint that the package was compiled for a more recent version of R. Re-installing quarto 1.2 from source appears to complete without error, but `quarto::quarto_version()` reports `Error in find_quarto(): unable to find quarto command line tools`.

I can confirm that quarto exists in both of these locations, but setting the environment variable `QUARTO_PATH` to either one of them does not clear the error:
`Sys.setenv(QUARTO_PATH="C:/ProgramData/R_Packages/quarto")`
`Sys.setenv(QUARTO_PATH="C:/Program Files/RStudio/bin/quarto")`

Any advice?!

**Comment by cderv** [comment]
> I was hoping to use quarto in a classroom

Are you referring to Quarto the CLI tool or only its R package counterpart (which only a wrapper for R) ? 

Installing quarto should not be tied to R versions - You can find it here: https://quarto.org/docs/download/ for latest version. 

RStudio IDE ships with a Quarto version too, but you are using an old one unfortunately. 
For RStudio IDE integration, we recommend v2022.07 though (https://quarto.org/docs/get-started/hello/rstudio.html). 
RStudio  2022.02.3 may not ship a quarto version - or if it is, it is best to use newer one. 


> Installing quarto from RStudio's package manager results in a complaint that the package was compiled for a more recent version of R

Installing the R package should work - it may warn because 4.2.2 is the current release for which window binary are built on CRAN, but it should not have impact on the package working. Anyway, installing from source is fine. 

> quarto::quarto_version() reports Error in find_quarto(): unable to find quarto command line tools.

Is quarto on your PATH ? What does `Sys.which("quarto")` in R returns ? 

Setting `QUARTO_PATH` should work for the R `package` only. If you want to call `quarto` in terminal window, it needs to be in PATH either global or set from R (`Sys.setenv(QUARTO_PATH="C:/ProgramData/R_Packages/quarto")`) and should work in IDE terminal.

> Sys.setenv(="C:/Program Files/RStudio/bin/quarto")

RStudio bundled Quarto should be found... 🤔 We may have fixed some issues though in v2022.07 as it was the version for the QUARTO v1 release. It would have been best to have this to work correctly with Quarto. Some features from IDE may not be working correctly with Quarto. You will probably need to use from Terminal or VS Code.

**Reply by johnghill** [reply]
Thank you @cderv 

> Are you referring to Quarto the CLI tool or only its R package counterpart (which only a wrapper for R) ?

The R package for use within RStudio IDE. 

> What does Sys.which("quarto") in R returns ?

I'll check this in the lab tomorrow, and report back.

> Setting `QUARTO_PATH` should work for the R `package` only.

Unfortunately this appears not to be the case, but I'll try `Sys.which("quarto")` as suggested.

My backup plan is to use RStudio Cloud but I was hoping to avoid the expense!

**Reply by cderv** [reply]
> My backup plan is to use RStudio Cloud but I was hoping to avoid the expense

It should work so we'll manage to configure correctly. 

Quarto needs to be installed, and put on PATH. After that, R should pick it up, unless an older config is blocking
